core/backend/src/TxnManager.ts

```diff
@@ -470,4 +482,14 @@ export class TxnManager {
   /** Query if there are un-saved or un-pushed local changes. */
   public get hasLocalChanges(): boolean { return this.hasUnsavedChanges || this.hasPendingTxns; }
 
+  /** Query local changes
+   * @param rootClassFilter Filter changes by a set of class and there derived classes.
+   * @param includeUnsavedChanges Include changes that has not saved yet in addition to saved changes.
+   *
+   * @beta
+  */
+  public queryLocalChanges(rootClassFilter: string[], includeUnsavedChanges: boolean = false): ChangeInstanceKey[] {
```

Comment:
Both arguments are optional (empty array means "no filter"), and additional arguments may be added in future.
Strongly consider
```
interface QueryLocalChangesArgs {
  rootClassFilter?: string;
  includeUnsavedChanges?: boolean;
}
```

Reply:
"filter" is always ambiguous as to whether it filters out things that **match** the criteria, or things that **don't** match.
Consider renaming to `includedClasses`.